website/docs/getting-started/project-walkthrough.md

```diff
@@ -0,0 +1,169 @@
+In CDK, all Resource Constructs must belong to a `Stack` - a `Stack` is a collection of Resources that are deployed and managed together.
+
+:::note
+
+You may be wondering why we have an `App` Construct when all Resource Constructs must belong to a `Stack`.
+
+An `App` may contain many stacks, but this is outside the scope of this document.
+You can read more about [`App`][cdk-app] and [`Stack`][cdk-stack] in the CDK documentation.
+
+:::
+
+Once we have an instance of `Stack`, we're ready to attach our first `Function`:
+
+```typescript
+const sayFunction = new Function(
+  stack,
+  "SayFunction",
+  async (event: { message: string }) => {
+    // This code will run on AWS Lambda
+    console.log(event.message);
+    return;
+  }
+);
+```
+
+This bit of code offers us our first glimpse at the magic of Functionless.
+If you've used CDK before, you're probably familiar with the strict seperation it imposes between infrastructure code and runtime code.
+
+:::info
+
+Functionless lets you write runtime code inline with the code that defines the infrastructure for your runtime code.
+
+:::
+
+:::note
+
+**Runtime code** refers the application code you write that will run after the infrastructure defined by your `Stack` has been deployed.
+
+**Infrastructure code** refers to the CDK code used to configure the services that host your runtime code as well as any supporting services used by your runtime code.
+
+:::
+
+This opens up a world of new possibilities for how our runtime code can interface with our infrastructure.
+We can see this in action in the `StepFunction` Construct definition:
+
+```typescript
+new StepFunction(stack, "Workflow", async (event: { name: string }) => {
+  // This will run in an AWS Step Function
+  await sayFunction({ message: `Hello ${event.name}` });
+});
+```
+
+Functionless is doing three important things for us in this block of code:
+
+1. Allowing us to call the `sayFunction` Construct as a typed TypeScript function.
+2. Recognizing that we've called an Lambda Function and granting Step Functions permissions to invoke that Lambda Function.
+3. Converting a block of TypeScript code into Amazon States Language (ASL) for us.
+
+We're able to call the `sayFunction` Construct we just defined as if it were a standard TypeScript function call.
+Additionally, the arguments and return type of `sayFunction` match what we defined above:
+
+```typescript
+type sayFunction = (event: { message: string }) => Promise<void>;
+```
+
+:::info
+
+Functionless uses type-safety to ensure the shape of your data is consistent throughout your application.
+Changing your `Function` definition to accept a `number` instead of `string` for `message` would raise a TypeScript compiler error.
+Functionless maintains types across service boundaries eliminating data shape consistency errors and saving countless hours of toil.
+
+:::
+
+"But wait!", you might say. "Don't I have to set up IAM permissions for this Step Function to call that Lambda?"
```

Comment:
This was mentioned in point 2 above.

Reply:
Hm yeah it does not work as well with the list at the top. Will rework.